### ISSUE TITLE
Podman Volume --type tmpfs seems to be a no-op

### ISSUE THREAD

**Issue by KCSesh** [containers/podman#10538]
Why do the podman examples for volume creation show: 

```
--opt device=tmpfs --opt type=tmpfs 
```

When those both appear to be No-Op flags for creating mounts? 
http://docs.podman.io/en/latest/markdown/podman-volume-create.1.html

tmpfs is a mount that is persisted in memory and is cleaned up when the container is killed. 

From testing creating volumes with --opt device=tmpfs --opt type=tmpfs, the contents are written to disk and not cleaned up when the container is stopped. 

1.
  
```
sudo podman --storage-driver vfs --root /data/podman/6845935107772988472/root --runroot /data/podman/6845935107772988472/run volume create --opt type=tmpfs --opt device=tmpfs --opt o=size=10m,uid=1005,gid=1007 volume-test
```

2. Container makes write to volume. and file is persisted to disk

3. 
```
sudo ls -la /data/podman/6845935107772988472/root/volumes/volume-test/_data
drwxrwxrwt+ 2 root                   root                     80 Jun  3 05:00 .
drwxrwx---+ 3 user1 user1 4096 Jun  3 04:50 ..
-rw-r--r--  1 user1 user1    0 Jun  3 05:00 new.txt
```

Seems the examples shouldn't include type=tmpfs at all as it leads to confusion. 

Docker seems to separate out Volume documentation from Tmpfs: 
https://docs.docker.com/storage/volumes/
https://docs.docker.com/storage/tmpfs/

**Comment by rhatdan** [comment]
Interested in opening a PR to fix the documentation?

**Comment by mheon** [comment]
This doesn't sound incorrect - the behavior here is expected. `/data/podman/6845935107772988472/root/volumes/volume-test/_data` should be a `tmpfs` mount. Files are being persisted to it and not cleaned up, yes - but when the system restarts, it will be wiped clean. If you did not want it persisted beyond a single container, you could use a tmpfs mount for the container in question - tmpfs volumes are mostly useful for being shared between containers, so it's expected that the contents won't disappear after a single container exits?

**Comment by mheon** [comment]
It that directory is not a tmpfs, this sounds like a bug.

**Comment by KCSesh** [comment]
**Looks like I was wrong. Apologies.** 

I created a volume of type tmpfs under root `/data/podman/6845935107772988472/root/volumes/volume-test/`

But then I mistakenly started the container under a separate root: `data/podman/<different>/root/` which resulted in the volume being created at runtime as a non-tmpfs. 

Upon testing again, this time with the correct tmpfs. 
I noticed when the container exists, the tmpfs clears out as expected. 


So it seems a volume can be created as a `tmpfs` and even be shared across multiple containers. As soon as no containers are running the tmpfs volume, the contents are removed.
